sklearn/compose/tests/test_column_transformer.py

```diff
@@ -2229,3 +2234,86 @@ def test_remainder_set_output():
     ct.set_output(transform="default")
     out = ct.fit_transform(df)
     assert isinstance(out, np.ndarray)
+
+
+# Metadata Routing Tests
+# ======================
+
+
+@pytest.mark.parametrize("method", ["transform", "fit_transform", "fit"])
+def test_routing_passed_metadata_not_supported(method):
+    """Test that the right error message is raised when metadata is passed while
+    not supported when `enable_metadata_routing=False`."""
+
+    X = np.array([[0, 1, 2], [2, 4, 6]]).T
+    y = [1, 2, 3]
+    trs = ColumnTransformer([("trans", Trans(), [0])]).fit(X, y)
+
+    with pytest.raises(
+        ValueError, match="is only supported if enable_metadata_routing=True"
+    ):
+        getattr(trs, method)([[1]], sample_weight=[1], prop="a")
+
+
+@pytest.mark.usefixtures("enable_slep006")
+@pytest.mark.parametrize("method", ["transform", "fit_transform", "fit"])
+def test_metadata_routing_for_column_transformer(method):
+    """Test that metadata is routed correctly for pipelines."""
+    X = np.array([[0, 1, 2], [2, 4, 6]]).T
+    y = [1, 2, 3]
+    registry = _Registry()
+    sample_weight, metadata = [1], "a"
+    trs = ColumnTransformer(
+        [
+            (
+                "trans",
+                ConsumingTransformer(registry=registry)
+                .set_fit_request(sample_weight=True, metadata=True)
+                .set_transform_request(sample_weight=True, metadata=True),
+                [0],
+            )
+        ]
+    )
+
+    if "fit" not in method:
+        trs.fit(X, y)
+
+    try:
+        getattr(trs, method)(X, y, sample_weight=sample_weight, metadata=metadata)
+    except TypeError:
+        # transform takes only X
+        getattr(trs, method)(X, sample_weight=sample_weight, metadata=metadata)
```

Comment:
I am wondering if you would simply have made:

```python
if method == "transform":
    trs.fit(X, y)
    getattr(trs, method)(X, sample_weight=sample_weight, metadata=metadata)
else:
    getattr(trs, method)(X, y, sample_weight=sample_weight, metadata=metadata)
```

I find it more explicit since it does not require any comment.